src/fields.jl

```diff
@@ -52,3 +52,231 @@ Field(LX, data, grid::Grid1D) = Field1D(LX, data, grid)
 Constructs a 2D field of `data` at location `(LX, LY)` on `grid`.
 """
 Field(LX, LY, data, grid::Grid2D) = Field2D(LX, LY, data, grid)
+
+############################################################################
+#-------------------------------Interpolation------------------------------#
+
+𝐼xᶠ(i::Int, f::Field1D{Face})   = (f.data[i] + f.data[i+1]) / 2
+𝐼xᶜ(i::Int, f::Field1D{Centre}) = (f.data[i] + f.data[i-1]) / 2
+
+𝐼xᶠᶜ(i::Int, j::Int, f::Field2D{Face, Centre})   = (f.data[i, j] + f.data[i+1, j]) / 2
+𝐼xᶜᶜ(i::Int, j::Int, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i-1, j]) / 2
+𝐼yᶜᶠ(i::Int, j::Int, f::Field2D{Centre, Face})   = (f.data[i, j+1] + f.data[i, j]) / 2
+𝐼yᶜᶜ(i::Int, j::Int, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i, j-1]) / 2
+
+"""
+    𝐼x!(output::Field1D{Centre}, input::Field1D{Face})
+
+Interpolates a 1D field of `data` from face to centre in x-direction.
+"""
+function 𝐼x!(output::Field1D{Centre}, input::Field1D{Face, Grid1D{Periodic}})
+    nx = input.grid.nx
+        
+    output.data[nx] = (input.data[1] + input.data[nx]) / 2
+    for i in 1:nx-1
+        output.data[i] = 𝐼xᶠ(i, input)
+    end
+end
+
+"""
+    𝐼x!(output::Field1D{Face}, input::Field1D{Centre})
+
+Interpolates a 1D field of `data` from centre to face in x-direction.
+"""
+function 𝐼x!(output::Field1D{Face}, input::Field1D{Centre, Grid1D{Periodic}})
+    nx = input.grid.nx
+        
+    output.data[1] = (input.data[1] + input.data[nx]) / 2
+    for i in 2:nx
+        output.data[i] = 𝐼xᶜ(i, input)
+    end
+end
+
+"""
+    𝐼x!(output::Field2D{Face, Centre}, input::Field2D{Centre, Centre})
+
+Interpolates a 2D field of `data` from centre to face in x-direction.
+"""
+function 𝐼x!(output::Field2D{Face, Centre}, input::Field2D{Centre, Centre, Grid2D{Periodic, Periodic}})
+    nx, ny = input.grid.nx, input.grid.ny
+    
+    for j in 1:ny
+        output.data[1, j] = (input.data[1, j] + input.data[nx, j]) / 2
+    end
+
+    for i in 2:nx
+        for j in 1:ny
+            output.data[i, j] = 𝐼xᶜᶜ(i, j, input)
+        end
+    end
```

Comment:
```suggestion
    for j in 1:ny, i in 2:nx
        output.data[i, j] = 𝐼xᶜᶜ(i, j, input)
    end
```

Julia is (column-major](https://en.wikipedia.org/wiki/Row-_and_column-major_order), like Matlab, FORTRAN. Looping within columns first makes a lot of difference. E.g. run this and look at the results:

```Julia
using BenchmarkTools

a = rand(200, 200);
b = rand(200, 200);
c = rand(200, 200);

function sum_columns_first!(c, a, b)
    for i=1:200, j=1:200
        c[i, j] = a[i, j] + b[i, j]
    end
end

function sum_rows_first!(c, a, b)
    for j=1:200, i=1:200
        c[i, j] = a[i, j] + b[i, j]
    end
end

@btime sum_columns_first!(c, a, b);

@btime sum_rows_first!(c, a, b);
```

Reply:
Note also how we can write nested `for` loops using `for i=1:100, j=1:100, k=1:100`.... Makes the code more readable I think.

Reply:
Yes I wanted to ask this, whether it's column major or row major. I'll keep it in mind.